src/js/_enqueues/admin/common.js

```diff
@@ -2222,6 +2222,14 @@ $( function( $ ) {
 	}
 })();
 
+// Attach a click event listener to the dismiss button for media upload errors.
+$( document ).on( 'click', 'button.button-link.dismiss', function( event ) {
+    event.preventDefault();
+    $( this ).parents( 'div.media-item' ).slideUp( 200, function() {
+        $( this ).remove();
+    });
+});
+
```

Comment:
I have mixed feelings about doing this in JQuery. Aren't we slowly moving towards Vanilla JS in WP? Imho adding new jQuery code should be avoided, but this just might be personal preference.